proxy/src/control/cache.rs

```diff
@@ -135,9 +156,12 @@ where
 
     pub fn clear<F>(&mut self, on_change: &mut F)
     where
-        F: FnMut((K, V), CacheChange),
+        F: for<'value> FnMut(CacheChange<'value, K, V>),
     {
-        self.update_intersection(IndexMap::new(), on_change)
```

Comment:
This function previously just called `update_intersection`. I've removed that call because `update_intersection` still requires the `V: Clone` bound, and I didn't want `clear` to require that `V: Clone`. This also means we no longer have to  construct a new empty `IndexMap` just to clear the cache.